message.go

```diff
@@ -61,6 +63,22 @@ func (this *CommandMsg) FromSocket(sock *Socket) {
 
 		sock.Page = page
 		sock.Server.Store.SetPage(sock) // set new page
+	case "setgroups":
+		grps, ok := this.Command["groups"]
+		grps = strings.TrimSpace(strings.Replace(grps, " ", "", -1))
+		groups := strings.Split(grps, ",")
+
+		if !ok || len(groups) == 0 {
+			return
+		}
+
+		// hack check for empty slice
+		if len(sock.Groups) != 0 {
```

Comment:
What I was trying to do is add groups to a single socket. So let's say you have your UID and you are apart of 3 different groups. I have mine and I'm in 2 groups. Someone sends you a message and to one of my groups. We both get that message kind of thing. Without too much knowledge of Redis I've been messing with just clearing them when setting new groups. Will gladly take recommendations on how you'd implement it.